File: frontend/src/types/lot.ts
export type LotStatus = 'for_sale' | 'reserved' | 'sold'

export interface Lot {
  id: number
  external_id: string
  set_id: number
  project_name: string
  address: string
  rooms: number
  area: string
  floor: number
  price: string
  price_base: string
  price_per_sqm: string | null
  status: LotStatus
  created_at: string
}

export type LotSortField = 'price' | 'price_per_sqm' | 'area' | 'created_at'
export type SortDirection = 'asc' | 'desc'

export interface LotListParams {
  project_name?: string
  rooms?: number
  price_per_sqm_min?: number
  price_per_sqm_max?: number
  status?: LotStatus
  sort_by?: LotSortField
  sort_dir?: SortDirection
  page?: number
  page_size?: number
}

export interface LotSet {
  id: number
  name: string
  uploaded_at: string
  lots_count: number
  is_active: boolean
}

export interface FeedUploadResult {
  lot_set: LotSet
  skipped_count: number
}

File: backend/app/schemas/lot.py
from datetime import datetime
from decimal import Decimal
from enum import Enum

from pydantic import BaseModel, ConfigDict, Field, HttpUrl

from app.models.lot import LotStatus


class LotRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: int
    external_id: str
    set_id: int
    project_name: str
    address: str
    rooms: int
    area: Decimal
    floor: int
    price: Decimal
    price_base: Decimal
    price_per_sqm: Decimal | None
    status: LotStatus
    created_at: datetime
    
class LotSortField(str, Enum):
    PRICE = "price"
    PRICE_PER_SQM = "price_per_sqm"
    AREA = "area"
    CREATED_AT = "created_at"
    
class SortDirection(str, Enum):
    ASC = "asc"
    DESC = "desc"
    
class LotListParams(BaseModel):
    project_name: str | None = None
    rooms: int | None = None
    price_per_sqm_min: Decimal | None = None
    price_per_sqm_max: Decimal | None = None
    status: LotStatus | None = None
    sort_by: LotSortField = LotSortField.CREATED_AT
    sort_dir: SortDirection = SortDirection.DESC
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)
    
class LotSetRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    
    id: int
    name: str
    uploaded_at: datetime
    lots_count: int
    is_active: bool
    
class FeedUploadResult(BaseModel):
    lot_set: LotSetRead
    skipped_count: int
    
class FeedUploadUrlRequest(BaseModel):
    url: HttpUrl
